File: Difficulty: Easy/Height of Binary Tree/height-of-binary-tree.cpp
/*
class Node {
public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = right = NULL;
    }
};
*/
class Solution {
  public:
    int height(Node* node) {
        // code here
        int ans =0;
        if(node == NULL) return 0;
        if(node->left ==NULL and node->right == NULL) return 0;
        
        queue<Node*> q;
        q.push(node);
        while(!q.empty()){
            int lvlSize = q.size();
            for(int i=0; i< lvlSize;i++){
                Node* top = q.front();
                q.pop();
                if(top->left) q.push(top->left);
                if(top->right) q.push(top->right);
            }
            ans++;
        }
        return ans-1;
    }
};

File: Difficulty: Easy/Height of Binary Tree/height-of-binary-tree.py
'''
# Node Class:
class Node:
    def _init_(self, val):
        self.data = val
        self.left = None
        self.right = None
'''

class Solution:
    def height(self, root):
        # code here
        
        def max_height(root):
            if root is None:
                return 0
            if root.left is None and root.right is None:
                return 0
            
            return 1 + max(max_height(root.left), max_height(root.right))
            
        
        return max_height(root)